sources/api/prairiedog/src/bootconfig.rs

```diff
@@ -0,0 +1,410 @@
+                values,
+            );
+        } else if !key.starts_with("kernel") && !key.starts_with("init") {
+            return error::UnsupportedBootConfigKeySnafu { key }.fail();
+        }
+    }
+
+    Ok(BootSettings {
+        kernel_parameters: if kernel_params.is_empty() {
+            None
+        } else {
+            Some(kernel_params)
+        },
+        init_parameters: if init_params.is_empty() {
+            None
+        } else {
+            Some(init_params)
+        },
+    })
+}
+
+/// Given a boot config string, deserialize it to `model::BootSettings` and then serialize it back
+/// out as a JSON string for sundog consumption
+fn boot_config_to_boot_settings_json(bootconfig_str: &str) -> Result<String> {
+    // We'll only send the setting if the existing boot config file fits our settings model
+    let boot_settings = parse_boot_config_to_boot_settings(bootconfig_str)?;
+    // sundog expects JSON-serialized output
+    serde_json::to_string(&boot_settings).context(error::OutputJsonSnafu)
+}
+
+#[cfg(test)]
+mod boot_settings_tests {
+    use crate::bootconfig::{
+        boot_config_to_boot_settings_json, serialize_boot_settings_to_boot_config,
+        DEFAULT_BOOTCONFIG_STR,
+    };
+    use maplit::hashmap;
+    use model::BootSettings;
+    use serde_json::json;
+    use serde_json::value::Value;
+    use std::convert::TryInto;
+
+    #[test]
+    fn boot_settings_to_string() {
+        let boot_settings = BootSettings {
+            kernel_parameters: Some(
+                hashmap! {
+                    "console" => vec!["ttyS1,115200n8", "tty0"],
+                }
+                .into_iter()
+                .map(|(k, v)| {
+                    (
+                        k.try_into().unwrap(),
+                        v.into_iter().map(|s| s.try_into().unwrap()).collect(),
+                    )
+                })
+                .collect(),
+            ),
+            init_parameters: Some(
+                hashmap! {
+                    "systemd.log_level" => vec!["debug"],
+                    "splash" => vec![""],
+                }
+                .into_iter()
+                .map(|(k, v)| {
+                    (
+                        k.try_into().unwrap(),
+                        v.into_iter().map(|s| s.try_into().unwrap()).collect(),
+                    )
+                })
+                .collect(),
+            ),
+        };
+        let output = serialize_boot_settings_to_boot_config(&boot_settings).unwrap();
+        // Sort the entries alphabetically to keep results consistent
+        let mut lines = output
+            .lines()
+            .map(|s| s.to_string())
+            .collect::<Vec<String>>();
+        lines.sort();
+        let output = lines.join("\n");
+        assert_eq!(
+            output,
+            r#"init.splash = ""
+init.systemd.log_level = "debug"
+kernel.console = "ttyS1,115200n8", "tty0""#
```

Comment:
nit: all of these tests would be more readable if given the same initial-newline and indent treatment as `STANDARD_BOOTCONFIG_STR` below.

Reply:
I'd have to use something like [this crate](https://docs.rs/unindent/latest/unindent/#) to ensure the leading whitespace is removed from the raw string.


`STANDARD_BOOTCONFIG_STR`'s situation is different because I `trim()` each line before parsing them down to K-Vs.

Reply:
I tried making this better with some extra `trim()`ing.